src/palladio/ShapeConverter.cpp

```diff
@@ -137,71 +137,68 @@ void forEachUVSet(const std::vector<GA_ROHandleV2D>& uvHandles, ConversionHelper
 	}
 }
 
-template <typename P>
-void convertPolygon(ConversionHelper& ch, const P& prim, const std::vector<GA_ROHandleV2D>& uvHandles) {
+void convertPolygon(ConversionHelper& ch, const GA_Primitive& prim, const std::vector<GA_ROHandleV2D>& uvHandles) {
 	const GA_Size vtxCnt = prim.getVertexCount();
-	uint32_t faceCnt = 1;
-	uint32_t holeCnt = 0;
-
-	std::optional<HoleConverter::FaceWithHoles> faceWithHoles;
-	if constexpr (std::is_same<P, GA_Primitive>()) {
-		auto const& face = static_cast<GEO_Face const&>(prim);
-		GeoFaceHoleSource geoFaceHoleConverter(face);
-		faceWithHoles = HoleConverter::extractHoles(geoFaceHoleConverter);
-		faceCnt = faceWithHoles->size();
-		holeCnt = faceCnt + 1; // as defined by PRT API: outer ring + hole rings + sentinel value
-	}
+
+	auto const& face = static_cast<GEO_Face const&>(prim);
+	GeoFaceHoleSource geoFaceHoleConverter(face);
+	HoleConverter::FaceWithHoles faceWithHoles = HoleConverter::extractHoles(geoFaceHoleConverter);
+	uint32_t faceCnt = faceWithHoles.size();
+	uint32_t holeCnt = faceCnt + 1; // as defined by PRT API: outer ring + hole rings + sentinel value
 
 	ch.indices.reserve(ch.indices.size() + vtxCnt);
 	ch.faceCounts.reserve(ch.faceCounts.size() + faceCnt);
 	ch.holes.reserve(ch.holes.size() + holeCnt);
 
-	if constexpr (std::is_same<P, GA_Primitive>()) {
-		for (HoleConverter::FaceOrHoleIndices const& faceOrHole : *faceWithHoles) {
-			ch.holes.push_back(ch.faceCounts.size());
-			ch.faceCounts.push_back(static_cast<uint32_t>(faceOrHole.size()));
-			std::for_each(faceOrHole.rbegin(), faceOrHole.rend(),
-			              [&prim, &ch](GA_Offset o) { ch.indices.push_back(prim.getPointIndex(o)); });
-		}
-
-		// required by PRT to delimit the holes belonging to a face
-		ch.holes.push_back(std::numeric_limits<uint32_t>::max());
-
-		forEachUVSet(uvHandles, ch, [&faceWithHoles, &prim](GA_ROHandleV2D const& uvh, UV& uvSet) {
-			for (HoleConverter::FaceOrHoleIndices const& faceOrHole : *faceWithHoles) {
-				for (GA_Offset j : faceOrHole) {
-					const auto v = uvh.get(prim.getVertexOffset(j));
-					uvSet.uvs.push_back(v.x());
-					uvSet.uvs.push_back(v.y());
-				}
-				std::vector<int32_t> indices(faceOrHole.size());
-				std::iota(indices.begin(), indices.end(), uvSet.idx.size());
-				uvSet.idx.insert(uvSet.idx.end(), indices.rbegin(), indices.rend());
-			}
-		});
+	for (HoleConverter::FaceOrHoleIndices const& faceOrHole : faceWithHoles) {
+		ch.holes.push_back(ch.faceCounts.size());
+		ch.faceCounts.push_back(static_cast<uint32_t>(faceOrHole.size()));
```

Comment:
why are we static casting only here? both use vector::size() no?
I didn't realize this the first round.

Reply:
GCC (at least) warns on implicit size_t -> uint32_t conversion, hence the `static_cast`